src/test/6_MainRegistry.t.sol

```diff
@@ -1038,3 +1038,142 @@ contract PricingLogicTest is MainRegistryTest {
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        (, uint128 oldExposure) = standardERC20PricingModule.exposure(address(link));
+
+        vm.startPrank(proxyAddr);
+        mainRegistry.batchProcessDeposit(assetAddresses, assetIds, assetAmounts);
+        vm.stopPrank();
+
+        (, uint128 newExposure) = standardERC20PricingModule.exposure(address(link));
+
+        assertEq(newExposure, oldExposure + amountLink);
+    }
+
+    function testRevert_batchProcessDeposit_delegateCall(uint128 amountLink) public {
+        address[] memory assetAddresses = new address[](1);
+        assetAddresses[0] = address(link);
+
+        uint256[] memory assetIds = new uint256[](1);
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        vm.startPrank(proxyAddr);
+        vm.expectRevert("Delegate calls not allowed.");
+        (bool success,) = address(mainRegistry).delegatecall(
+            abi.encodeWithSignature(
+                "batchProcessDeposit(address[] calldata,uint256[] calldata,uint256[] calldata)",
+                assetAddresses,
+                assetIds,
+                assetAmounts
+            )
+        );
+        vm.stopPrank();
+
+        success; //avoid warning
+    }
+
+    function testSuccess_batchProcessWithdrawal_directCall(uint128 amountLink) public {
+        address[] memory assetAddresses = new address[](1);
+        assetAddresses[0] = address(link);
+
+        uint256[] memory assetIds = new uint256[](1);
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        (, uint128 oldExposure) = standardERC20PricingModule.exposure(address(link));
+
+        vm.startPrank(proxyAddr);
+        mainRegistry.batchProcessDeposit(assetAddresses, assetIds, assetAmounts);
+        vm.stopPrank();
+
+        (, uint128 newExposure) = standardERC20PricingModule.exposure(address(link));
+
+        assertEq(newExposure, oldExposure + amountLink);
+
+        vm.startPrank(proxyAddr);
+        mainRegistry.batchProcessWithdrawal(assetAddresses, assetAmounts);
+        vm.stopPrank();
+
+        (, uint128 endExposure) = standardERC20PricingModule.exposure(address(link));
+
+        assertEq(endExposure, oldExposure);
+    }
+
+    function testRevert_batchProcessWithdrawal_delegateCall(uint128 amountLink) public {
+        address[] memory assetAddresses = new address[](1);
+        assetAddresses[0] = address(link);
+
+        uint256[] memory assetIds = new uint256[](1);
+        assetIds[0] = 0;
+
+        uint256[] memory assetAmounts = new uint256[](1);
+        assetAmounts[0] = amountLink;
+
+        vm.startPrank(proxyAddr);
+        vm.expectRevert("Delegate calls not allowed.");
+        (bool success,) = address(mainRegistry).delegatecall(
```

Comment:
```suggestion
        address(mainRegistry).delegatecall(
```

Reply:
will throw a 'return of delegatecall not evaluate' or something similar